integration/metrics_test.go

```diff
@@ -192,3 +201,189 @@ disable_verification = %v
+		}
+		return []byte(testutil.ApplyTextTemplate(t, `
+version = 2
+[plugins."io.containerd.snapshotter.v1.soci"]
+root_path = "/var/lib/soci-snapshotter-grpc/"
+disable_verification = {{.DisableVerification}}
+[plugins."io.containerd.snapshotter.v1.soci".blob]
+check_always = true
+[debug]
+format = "json"
+level = "debug"
+{{.AdditionalConfig}}
+`, struct {
+			DisableVerification bool
+			AdditionalConfig    string
+		}{
+			DisableVerification: disableVerification,
+			AdditionalConfig:    additionalConfig,
+		}))
+	}
+	getSnapshotterConfigYaml := func(disableVerification bool) []byte {
+		return []byte(fmt.Sprintf(`
+metrics_address = "localhost:6060"
+disable_verification = %v
+[fuse]
+log_fuse_operations = true
+`, disableVerification))
+	}
+
+	sh, done := newShellWithRegistry(t, regConfig)
+	defer done()
+
+	if err := testutil.WriteFileContents(sh, defaultContainerdConfigPath, getContainerdConfigYaml(false), 0600); err != nil {
+		t.Fatalf("failed to write %v: %v", defaultContainerdConfigPath, err)
+	}
+	if err := testutil.WriteFileContents(sh, defaultSnapshotterConfigPath, getSnapshotterConfigYaml(false), 0600); err != nil {
+		t.Fatalf("failed to write %v: %v", defaultSnapshotterConfigPath, err)
+	}
+
+	testCases := []struct {
+		name                       string
+		image                      string
+		indexDigestFn              func(*testing.T, *shell.Shell, imageInfo) string
+		expectFuseOperationFailure bool
+	}{
+		{
+			name:  "image with valid ztocs and index results in fuse operation failures",
+			image: rabbitmqImage,
+			indexDigestFn: func(t *testing.T, sh *shell.Shell, image imageInfo) string {
+				return optimizeImageWithOpts(sh, image, 1<<22, 0)
+			},
+			expectFuseOperationFailure: false,
+		},
+		{
+			name:  "image with valid-formatted but invalid-data ztocs results in fuse operation failures",
+			image: rabbitmqImage,
+			indexDigestFn: func(t *testing.T, sh *shell.Shell, image imageInfo) string {
+				indexDigest := optimizeImageWithOpts(sh, image, 1<<22, 0)
+				return buildIndexByManipulatingZtocData(t, sh, indexDigest)
+			},
+			expectFuseOperationFailure: true,
+		},
+	}
+
+	// checkFuseOperationFailureMetrics checks if output from metrics endpoint includes
+	// any fuse operation failure metrics
+	checkFuseOperationFailureMetrics := func(t *testing.T, output string, expectFuseOperationFailure bool) {
+		metricCountSum := 0
+
+		lines := strings.Split(output, "\n")
+		for _, line := range lines {
+			// filter out non-fuse and fuse_mount_failure_count metrics
+			if !strings.Contains(line, "fuse") || strings.Contains(line, commonmetrics.FuseMountFailureCount) {
+				continue
+			}
+
+			parts := strings.Split(line, " ")
+			if metricCount, err := strconv.Atoi(parts[len(parts)-1]); err == nil && metricCount != 0 {
+				if !expectFuseOperationFailure {
+					t.Fatalf("unexpected fuse operation failure: %s", line)
+				}
+				t.Logf("fuse operation failure metric: %s", line)
```

Comment:
Do we need this log?

Reply:
I think we need. if we are not expecting fuse errors but there are errors from metrics output, we want to know what the  fuse error is